src/components/PostDetails.tsx

```diff
@@ -1,99 +1,53 @@
 import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
```

Comment:
Ensure that `useDispatch` and `useSelector` are correctly imported from 'react-redux'.